src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs

```diff
@@ -89,5 +89,51 @@ public static string GetDotnetExeDirectory(Func<string, string> getEnvironmentVa
             var environmentProvider = new EnvironmentProvider(getEnvironmentVariable);
             return environmentProvider.GetDotnetExeDirectory();
         }
+
+        public string GetUserHomeDirectory()
+        {
+            var home = _getEnvironmentVariable("DOTNET_CLI_HOME");
+            if (string.IsNullOrEmpty(home))
+            {
+                home = _getEnvironmentVariable(Interop.RunningOnWindows ? "USERPROFILE" : "HOME");
+                if (string.IsNullOrEmpty(home))
+                {
+                    throw new Exception("Unable to resolve user home");
+                }
+            }
+
+            return home;
+        }
+
+        public static string GetUserHomeDirectory(Func<string, string> getEnvironmentVariable = null)
+        {
+            if (getEnvironmentVariable == null)
+            {
+                getEnvironmentVariable = Environment.GetEnvironmentVariable;
+            }
+            var environmentProvider = new EnvironmentProvider(getEnvironmentVariable);
+            return environmentProvider.GetUserHomeDirectory();
+        }
+
+        public string GetTempDirectory()
+        {
+            var tmp = _getEnvironmentVariable(Interop.RunningOnWindows ? "TEMP" : "TMPDIR");
```

Comment:
Why are we using environment variables? Path.GetTempPath works on both Windows and Linux, and it implements a set of fallback rules. For example, on Linux it falls back to /tmp if TMPDIR doesn't exist, but here we would throw an exception.